test/extended/util/oauthserver/oauthserver.go

```diff
@@ -152,51 +143,87 @@ func DeployOAuthServer(oc *exutil.CLI, idps []osinv1.IdentityProvider, configMap
 	if err != nil {
-		return nil, cleanups, err
+		return nil, cleanupFunc, err
 	}
 
 	// prepare the pod def, create secrets and CMs
 	oauthServerPod, err := oauthServerPod(configMaps, secrets, image)
 	if err != nil {
-		return nil, cleanups, err
+		return nil, cleanupFunc, err
 	}
 
 	// finally create the oauth server, wait till it starts running
-	if _, err := coreClient.Pods(oc.Namespace()).Create(oauthServerPod); err != nil {
-		return nil, cleanups, err
+	if _, err := kubeClient.CoreV1().Pods(oc.Namespace()).Create(oauthServerPod); err != nil {
+		return nil, cleanupFunc, err
+	}
+
+	if err := waitForOAuthServerReady(oc); err != nil {
+		return nil, cleanupFunc, err
+	}
+
+	oauthClient, err := createOAuthClient(oc, routeURL)
+	if err != nil {
+		return nil, cleanupFunc, err
+	}
+	cleanupFuncs = append(cleanupFuncs, func() {
+		_ = oc.AsAdmin().Run("delete").Args("oauthclients.oauth.openshift.io", oauthClient.Name).Execute()
+	})
+
+	newRequestTokenOptionFunc := func(username, password string) *tokencmd.RequestTokenOptions {
+		return newRequestTokenOptions(restclient.AnonymousClientConfig(oc.AdminConfig()), routeURL, oc.Namespace(), username, password)
 	}
 
-	err = wait.PollImmediate(1*time.Second, 45*time.Second, func() (bool, error) {
+	return newRequestTokenOptionFunc, cleanupFunc, nil
+}
+
+func waitForOAuthServerReady(oc *exutil.CLI) error {
+	if err := exutil.WaitForUserBeAuthorized(oc, "system:serviceaccount:"+oc.Namespace()+":e2e-oauth", "get", "oauthclients.oauth.openshift.io"); err != nil {
+		return err
+	}
+	if err := waitForOAuthServerRouteReady(oc); err != nil {
+		return err
+	}
+	return waitForOAuthServerPodReady(oc)
+}
+
+func waitForOAuthServerPodReady(oc *exutil.CLI) error {
+	return wait.PollImmediate(1*time.Second, 45*time.Second, func() (bool, error) {
 		pod, err := oc.AdminKubeClient().CoreV1().Pods(oc.Namespace()).Get("test-oauth-server", metav1.GetOptions{})
 		if err != nil {
 			return false, err
 		}
 		if !exutil.CheckPodIsReady(*pod) {
 			return false, nil
 		}
-
 		return true, nil
 	})
+}
 
+func waitForOAuthServerRouteReady(oc *exutil.CLI) error {
+	route, err := oc.AdminRouteClient().RouteV1().Routes(oc.Namespace()).Get(RouteName, metav1.GetOptions{})
 	if err != nil {
-		return nil, cleanups, err
+		return err
 	}
-
-	if err = createOAuthClient(oc, routeURL); err != nil {
-		return nil, cleanups, err
-	}
-	tokenReqOptions, err := getTokenOpts(oc.AdminConfig(), routeURL, oc.Namespace())
+	request, err := http.NewRequest(http.MethodHead, fmt.Sprintf("https://%s/healthz", route.Spec.Host), nil)
 	if err != nil {
-		return nil, cleanups, err
+		return err
 	}
-
-	return tokenReqOptions, cleanups, nil
+	c := 0
+	return wait.PollImmediate(time.Second, time.Minute, func() (done bool, err error) {
+		c++
+		transport, err := restclient.TransportFor(restclient.AnonymousClientConfig(oc.AdminConfig()))
+		if err != nil {
+			return false, err
+		}
+		response, err := transport.RoundTrip(request)
```

Comment:
explicitly ignore error if you do not want to check it

Reply:
Done